core/includes/form.inc

```diff
@@ -4750,6 +4750,8 @@ function theme_textarea($variables) {
 function theme_password($variables) {
   $element = $variables['element'];
   $element['#attributes']['type'] = 'password';
+  $element['#attributes']['spellcheck'] = !empty($element['#attributes']['spellcheck']) ? $element['#attributes']['spellcheck'] : 'false';
```

Comment:
This is the low-level FAPI password theming function. This checks to see if we have already set  the attribute`spellcheck` in the form definition. If so, it respects that. If not, it adds `'false'` by default.